core/src/main/java/org/apache/calcite/jdbc/JavaTypeFactoryImpl.java

```diff
@@ -243,11 +243,24 @@ public RelDataType toSql(RelDataType type) {
   /** Converts a type in Java format to a SQL-oriented type. */
   public static RelDataType toSql(final RelDataTypeFactory typeFactory,
       RelDataType type) {
+    return toSql(typeFactory, type, true);
+  }
+
+  private static RelDataType toSql(final RelDataTypeFactory typeFactory,
+      RelDataType type, boolean isRootType) {
```

Comment:
I would rename `isRootType` to `mustConsiderNullability` or `mustSetNullability` or something similar.